### PR TITLE
Internal types, methods and variables renames in :workers

### PR DESCRIPTION
**Pull request by eskatos**

### CODE REVIEW

File: subprojects/workers/src/main/java/org/gradle/workers/internal/WorkerDaemonStarter.java [hunk 1]
```diff
@@ -54,7 +54,7 @@ public WorkerDaemonClient startDaemon(Class<? extends WorkerDaemonProtocol> serv
         javaCommand.setMaxHeapSize(forkOptions.getMaxHeapSize());
         javaCommand.setJvmArgs(forkOptions.getJvmArgs());
         javaCommand.setWorkingDir(workingDir);
-        WorkerDaemonWorker worker = builder.build();
+        WorkerDaemonProcess worker = builder.build();
```
Comment (ghale) [inline subprojects/workers/src/main/java/org/gradle/workers/internal/WorkerDaemonStarter.java]: Let's rename this to `workerDaemonProcess`.

File: subprojects/workers/src/main/java/org/gradle/workers/internal/WorkerDaemonStarter.java [hunk 2]
```diff
@@ -41,10 +41,10 @@ public WorkerDaemonStarter(WorkerProcessFactory workerFactory, StartParameter st
         this.buildOperationExecutor = buildOperationExecutor;
     }
 
-    public WorkerDaemonClient startDaemon(Class<? extends WorkerDaemonProtocol> serverImplementationClass, File workingDir, DaemonForkOptions forkOptions) {
+    public WorkerDaemonClient startDaemon(Class<? extends WorkerProtocol> workerProtocolImplementationClass, File workingDir, DaemonForkOptions forkOptions) {
         LOG.debug("Starting Gradle worker daemon with fork options {}.", forkOptions);
         Timer clock = Timers.startTimer();
-        MultiRequestWorkerProcessBuilder<WorkerDaemonWorker> builder = workerFactory.multiRequestWorker(WorkerDaemonWorker.class, WorkerDaemonProtocol.class, serverImplementationClass);
+        MultiRequestWorkerProcessBuilder<WorkerDaemonProcess> builder = workerFactory.multiRequestWorker(WorkerDaemonProcess.class, WorkerProtocol.class, workerProtocolImplementationClass);
```
Comment (ghale) [inline subprojects/workers/src/main/java/org/gradle/workers/internal/WorkerDaemonStarter.java]: Or rather `workerDaemonProcessFactory`

File: subprojects/workers/src/main/java/org/gradle/workers/internal/WorkersServices.java [hunk 3]
```diff
@@ -58,16 +58,16 @@ WorkerDaemonClientsManager createWorkerDaemonClientsManager(WorkerProcessFactory
...
+        WorkerDaemonFactory createWorkerDaemonFactory(WorkerDaemonClientsManager workerDaemonClientsManager, MemoryManager memoryManager, BuildOperationWorkerRegistry buildOperationWorkerRegistry, BuildOperationExecutor buildOperationExecutor) {
+            return new WorkerDaemonFactory(workerDaemonClientsManager, memoryManager, buildOperationWorkerRegistry, buildOperationExecutor);
         }
 
-        WorkerExecutor createWorkerDaemonExecutor(WorkerDaemonManager workerDaemonFactory, InProcessCompilerDaemonFactory inProcessWorkerDaemonFactory, FileResolver fileResolver, ExecutorFactory executorFactory, BuildOperationWorkerRegistry buildOperationWorkerRegistry, BuildOperationExecutor buildOperationExecutor, AsyncWorkTracker asyncWorkTracker) {
+        WorkerExecutor createWorkerExecutor(WorkerDaemonFactory workerDaemonFactory, InProcessWorkerFactory inProcessWorkerDaemonFactory, FileResolver fileResolver, ExecutorFactory executorFactory, BuildOperationWorkerRegistry buildOperationWorkerRegistry, BuildOperationExecutor buildOperationExecutor, AsyncWorkTracker asyncWorkTracker) {
```
Comment (ghale) [inline subprojects/workers/src/main/java/org/gradle/workers/internal/WorkersServices.java]: `inProcessWorkerDaemonFactory` -> `inProcessWorkerFactory`

File: subprojects/workers/src/test/groovy/org/gradle/workers/internal/WorkerDaemonClientTest.groovy [hunk 4]
```diff
@@ -38,13 +38,13 @@ class WorkerDaemonClientTest extends Specification {
     }
 
     def "underlying worker is executed when client is executed"() {
-        def workerDaemonWorker = Mock(WorkerDaemonWorker)
+        def workerDaemonWorker = Mock(WorkerDaemonProcess)
```
Comment (ghale) [inline subprojects/workers/src/test/groovy/org/gradle/workers/internal/WorkerDaemonClientTest.groovy]: `workerDaemonWorker` -> `workerDaemonProcess`

File: subprojects/workers/src/test/groovy/org/gradle/workers/internal/WorkerDaemonClientTest.groovy [hunk 5]
```diff
@@ -86,13 +86,13 @@ class WorkerDaemonClientTest extends Specification {
     def "build worker operation is finished even if worker fails"() {
         def operation = Mock(Operation)
         def completion = Mock(Completion)
-        def workerDaemonWorker = Mock(WorkerDaemonWorker)
+        def workerDaemonWorker = Mock(WorkerDaemonProcess)
```
Comment (ghale) [inline subprojects/workers/src/test/groovy/org/gradle/workers/internal/WorkerDaemonClientTest.groovy]: `workerDaemonWorker` -> `workerDaemonProcess`

File: subprojects/workers/src/test/groovy/org/gradle/workers/internal/WorkerDaemonClientTest.groovy [hunk 6]
```diff
@@ -105,10 +105,10 @@ class WorkerDaemonClientTest extends Specification {
     }
 
     WorkerDaemonClient client() {
-        return client(Mock(WorkerDaemonWorker))
+        return client(Mock(WorkerDaemonProcess))
     }
 
-    WorkerDaemonClient client(WorkerDaemonWorker workerDaemonWorker) {
+    WorkerDaemonClient client(WorkerDaemonProcess workerDaemonWorker) {
```
Comment (ghale) [inline subprojects/workers/src/test/groovy/org/gradle/workers/internal/WorkerDaemonClientTest.groovy]: `workerDaemonWorker` -> `workerDaemonProcess`

File: subprojects/workers/src/test/groovy/org/gradle/workers/internal/WorkerDaemonFactoryTest.groovy [hunk 7]
```diff
@@ -33,25 +33,25 @@ class WorkerDaemonManagerTest extends Specification {
     def workerOperation = Mock(Operation)
     def buildOperation = Mock(BuildOperationExecutor.Operation)
 
-    @Subject manager = new WorkerDaemonManager(clientsManager, memoryManager, buildOperationWorkerRegistry, buildOperationExecutor)
+    @Subject manager = new WorkerDaemonFactory(clientsManager, memoryManager, buildOperationWorkerRegistry, buildOperationExecutor)
```
Comment (ghale) [inline subprojects/workers/src/test/groovy/org/gradle/workers/internal/WorkerDaemonFactoryTest.groovy]: `manager` -> `factory`

**Comment by eskatos** [comment]
@ghale, thanks for the thorough review, I applied your comments, could you have another look?

**Pull request merged**